tiled/client/base.py

```diff
@@ -1,9 +1,94 @@
+import copy
 import importlib
+import time
 
 from ..utils import UNCHANGED, DictView, ListView, OneShotCachedMap
 from .cache import Revalidate, verify_cache
 
 
+class Metadata_Revisions:
+    def __init__(self, context, path):
+        self._cached_len = None
+        self.context = context
+        self._path = path
+
+    def __len__(self):
+
+        LENGTH_CACHE_TTL = 1  # second
+
+        now = time.monotonic()
+        if self._cached_len is not None:
+            length, deadline = self._cached_len
+            if now < deadline:
+                # Used the cached value and do not make any request.
+                return length
+
+        path = (
+            "/node/revisions"
+            + "".join(f"/{part}" for part in self.context.path_parts)
+            + "".join(f"/{part}" for part in (self._path or [""]))
+        )
+
+        content = self.context.get_json(
+            path, params={"page[offset]": 0, "page[limit]": 0}
+        )
+        length = content["meta"]["count"]
+        self._cached_len = (length, now + LENGTH_CACHE_TTL)
+        return length
+
+    def __getitem__(self, item_):
+        self._cached_len = None
+
+        path = (
+            "/node/revisions"
+            + "".join(f"/{part}" for part in self.context.path_parts)
+            + "".join(f"/{part}" for part in (self._path or [""]))
+        )
+
+        if isinstance(item_, int):
+            offset = item_
+            limit = 1
+
+            return self.context.get_json(
+                path, params={"page[offset]": offset, "page[limit]": limit}
+            )
+
+        elif isinstance(item_, slice):
+            offset = item_.start
+            if offset is None:
+                offset = 0
+
+            if item_.stop is None:
+                params = f"?page[offset]={offset}"
+            else:
+                limit = item_.stop - offset
+                params = f"page[offset]={offset}&page[limit]={limit}"
+
+            next_page = path + params
+            result = None
```

Comment:
Since we now plan to return the content of `"data"`, I think this can be initialized as a list.